runtime/container.go

```diff
@@ -173,79 +168,6 @@ func (c *container) Labels() []string {
 	return c.labels
 }
 
-func (c *container) Start(checkpoint string, s Stdio) (Process, error) {
-	processRoot := filepath.Join(c.root, c.id, InitProcessID)
-	if err := os.Mkdir(processRoot, 0755); err != nil {
-		return nil, err
-	}
-	cmd := exec.Command("containerd-shim",
-		c.id, c.bundle,
-	)
-	cmd.Dir = processRoot
-	cmd.SysProcAttr = &syscall.SysProcAttr{
-		Setpgid: true,
-	}
-	spec, err := c.readSpec()
-	if err != nil {
-		return nil, err
-	}
-	config := &processConfig{
-		checkpoint:  checkpoint,
-		root:        processRoot,
-		id:          InitProcessID,
-		c:           c,
-		stdio:       s,
-		spec:        spec,
-		processSpec: spec.Process,
-	}
-	p, err := newProcess(config)
-	if err != nil {
-		return nil, err
-	}
-	if err := cmd.Start(); err != nil {
-		return nil, err
-	}
-	if _, err := p.getPid(); err != nil {
-		return p, nil
-	}
-	c.processes[InitProcessID] = p
-	return p, nil
-}
-
-func (c *container) Exec(pid string, spec specs.Process, s Stdio) (Process, error) {
-	processRoot := filepath.Join(c.root, c.id, pid)
-	if err := os.Mkdir(processRoot, 0755); err != nil {
-		return nil, err
-	}
-	cmd := exec.Command("containerd-shim",
-		c.id, c.bundle,
-	)
-	cmd.Dir = processRoot
-	cmd.SysProcAttr = &syscall.SysProcAttr{
-		Setpgid: true,
-	}
-	config := &processConfig{
-		exec:        true,
-		id:          pid,
-		root:        processRoot,
-		c:           c,
-		processSpec: spec,
-		stdio:       s,
-	}
-	p, err := newProcess(config)
-	if err != nil {
-		return nil, err
-	}
-	if err := cmd.Start(); err != nil {
-		return nil, err
-	}
-	if _, err := p.getPid(); err != nil {
-		return p, nil
-	}
-	c.processes[pid] = p
-	return p, nil
-}
-
 func (c *container) readSpec() (*platformSpec, error) {
```

Comment:
Here too, `platformSpec` should be made public within the runtime package.

I'll do a patch for Linux. I unfortunately haven't managed to get a working TP4 build environment yet, so I can't do the changes for windows :-/